bulk_indexer.go

```diff
@@ -709,14 +730,22 @@ func indexnth(s []byte, nth int, sep rune) int {
 	})
 }
 
-type errorFlushFailed struct {
+type ErrorFlushFailed struct {
 	resp        string
 	statusCode  int
 	tooMany     bool
 	clientError bool
 	serverError bool
 }
 
-func (e errorFlushFailed) Error() string {
+func (e *ErrorFlushFailed) StatusCode() int {
+	return e.statusCode
+}
+
+func (e *ErrorFlushFailed) ResponseBody() string {
+	return e.resp
+}
+
+func (e *ErrorFlushFailed) Error() string {
```

Comment:
nit: can we keep these as value receivers? The added benefit is we can skip the pointer gymnastics in returns and errors.As